content/kb/tutorials/databases/supabase.md

```diff
@@ -108,23 +123,14 @@ Copy the code below to your Streamlit app and run it.
 # streamlit_app.py
 
 import streamlit as st
-from supabase import create_client, Client
+from st_supabase_connection import SupabaseConnection
 
 # Initialize connection.
-# Uses st.cache_resource to only run once.
-@st.cache_resource
-def init_connection():
-    url = st.secrets["supabase_url"]
-    key = st.secrets["supabase_key"]
-    return create_client(url, key)
-
-supabase = init_connection()
+supabase = st.experimental_connection("supabase",type=SupabaseConnection)
 
 # Perform query.
-# Uses st.cache_data to only rerun when the query changes or after 10 min.
-@st.cache_data(ttl=600)
 def run_query():
-    return supabase.table("mytable").select("*").execute()
+    return supabase.query("*", table="mytable", ttl=600).execute()
 
 rows = run_query()
```

Comment:
request that we update this section to roughly the following, unless there's a good reason to keep it this way:

```python
conn = st.experimental_connection("supabase", type=SupabaseConnection)

# Perform query.
rows = conn.query("*", table="mytable", ttl="10m").execute()
```

Reply:
Much, better! Thanks 😄